apis/core/v1alpha1/flagsourceconfiguration_types.go

```diff
@@ -95,68 +95,128 @@ type FlagSourceConfigurationSpec struct {
 	// +optional
 	DefaultSyncProvider SyncProviderType `json:"defaultSyncProvider"`
 
+	// Sources defines the syncProviders and associated configuration to be applied to the sidecar
+	// +optional
+	Sources []Source `json:"sources"`
```

Comment:
We may want to make this field required, and also force at least 1 item in the array. As is, if you don't put anything here, flagd attempts to start with no args and it results in a CrashLoopBackoff ( @beeme1mr  found this in his testing).

`// +kubebuilder:validation:MinItems` seems to exist for this purpose, see [here](https://book.kubebuilder.io/reference/markers/crd-validation.html).

Reply:
The unique validator may also make sense.

Reply:
Ive introduced the `MinItems` tag, however, adding the unique validator cannot also be added as `the runtime complexity becomes quadratic`. If we want to introduce this change we can always introduce the logic into the validation webhook